app/uk/gov/hmrc/gform/models/gform/FormValidationOutcome.scala

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.models.gform
+import uk.gov.hmrc.gform.sharedmodel.form.{ FormData, ValidationResult }
+import uk.gov.hmrc.gform.validation.ValidationUtil.ValidatedType
+
+case class FormValidationOutcome(isValid: Boolean, formData: FormData, validatedType: ValidatedType[ValidationResult])
```

Comment:
maybe replace `isValid` with an ADT but if takes more effort than gain don't worry

Reply:
I could do this as an exercise but I really need to get this approved as I need to make some changes to master.